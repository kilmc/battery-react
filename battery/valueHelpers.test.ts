// eslint-env jest, node

import {
  baseRemify,
  baseScaler,
  opacify,
  baseColorHex,
  baseLengths,
  baseColors,
  baseIntegers,
  baseKeywords,
  defaultValue,
  Config
} from './valueHelpers';

const config: Config = {
  baseFontSize: 10,
  baseUnit: 6,
  systemColors: {
    'hot-pink': '#FF0099',
    blue: '#0000FF'
  }
};

const remify = baseRemify(config);
const scale = baseScaler(config);
const colorHex = baseColorHex(config);

const lengths = baseLengths(config);
const colors = baseColors(config);
const integers = baseIntegers(config);
const keywords = baseKeywords(config);

describe('remify', () => {
  it('divides integer by baseFontSize', () => {
    expect(remify(24)).toEqual(2.4);
  });
});

describe('scaler', () => {
  it('multiplies integer by baseUnitSize', () => {
    expect(scale(4)).toEqual(24);
  });
});

describe('opacify', () => {
  it('divides integer by 100', () => {
    expect(opacify(10)).toEqual(0.1);
  });
});

describe('colorHex', () => {
  it('returns HEX value from color name', () => {
    expect(colorHex('hot-pink')).toEqual('#FF0099');
  });
});

describe('Lengths', () => {
  it('matches snapshot', () => {
    expect(
      lengths({
        values: [1, 2, 3, 4],
        lengthUnit: 'rem',
        valueConverters: [remify, scale]
      })
    ).toMatchSnapshot();
  });
});

describe('colors', () => {
  it('matches snapshot', () => {
    expect(colors(['blue'])).toMatchSnapshot();
  });
});

describe('integers', () => {
  it('matches snapshot', () => {
    expect(integers([1, 2, 3, 4])).toMatchSnapshot();
  });
});

describe('keywords', () => {
  it('matches snapshot', () => {
    expect(
      keywords({
        block: 'block',
        hide: 'none'
      })
    ).toMatchSnapshot();
  });
});

describe('defaultValue', () => {
  it('matches snapshot', () => {
    expect(
      defaultValue(`0.1rem solid ${colorHex('hot-pink')}`)
    ).toMatchSnapshot();
  });
});
