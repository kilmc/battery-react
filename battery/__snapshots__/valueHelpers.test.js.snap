// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lengths matches snapshot 1`] = `
Object {
  "1": "0.6rem",
  "2": "1.2rem",
  "3": "1.8rem",
  "4": "2.4rem",
}
`;

exports[`colors matches snapshot 1`] = `
Object {
  "blue": "#0000FF",
}
`;

exports[`defaultValue matches snapshot 1`] = `
Object {
  "": "0.1rem solid #FF0099",
}
`;

exports[`integers matches snapshot 1`] = `
Object {
  "1": 1,
  "2": 2,
  "3": 3,
  "4": 4,
}
`;

exports[`keywords matches snapshot 1`] = `
Object {
  "block": "block",
  "hide": "none",
}
`;
